### ISSUE TITLE
Create (optional?) xpath preprocessor

### ISSUE THREAD

**Issue by fprochazka** [kaja47/Matcher#12]
Hi, [the preprocessor](https://github.com/paranoiq/dogma/blob/master/Dogma/Dom/QueryEngine.php#L27) by @paranoiq has very nice features, would you consider also creating preprocessor, or including parts if his in this lib? Some of the aliases are very nice, native xpath is very verbose.

**Comment by kaja47** [comment]
Well, I'm not too keen on rewriting XPaths with regexps.
But as of current master it's easy to plug in custom preprocessor.

```php
$context = new MatcherContext(null, [], 'Symfony\Component\CssSelector\CssSelector::toXpath');

Matcher::multi('div.article > h2')->fromHtml($context);
```
On the other hand, I have complete XPath expression parser and it could be modified and used as a preprocessor.

**Comment by fprochazka** [comment]
Yeah, optionally :) You must admit some of those aliases are really nice.
